shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -43,7 +43,7 @@
  * Incompatible changes should instead create a new WireCommand object.
  */
 public final class WireCommands {
-    public static final int WIRE_VERSION = 3;
+    public static final int WIRE_VERSION = 4;
```

Comment:
I understand that we are bumping up the wire version because we are adding a new message, but what happens if this message hits a client that doesn't understand it? We should only send such a message in the case the client supports it.

Reply:
I thought we don't allow the connection to go through if the client doesn't support the version? The Wire version is not stored anywhere server-side if I understand the code correctly, so we can't really do what you're suggesting ATM.

Reply:
> I thought we don't allow the connection to go through if the client doesn't support the version?

That's correct, but the problem with this configuration is that the server is going to it is compatible because the lowest compatible version is 1. We could make the oldest compatible be 4, which would make all previous version incompatible, but according to our guidelines for protocol changes, new replies should be sent selectively according to version. Let me review it to see if I can suggest something more concrete here.

Reply:
We don't have this pattern or support for this anywhere in the protocol. I don't want to add that feature it as part of this PR as it's unrelated.

Reply:
If the client gets an unexpected reply type, it will throw an exception when decoding it which will result in the exception being logged and the client closing the connection. 
Given that the response type is an error that the client cannot do anything about, this is as good as we can reasonably do. So I don't see any need to declare this version incompatible. We can bump this wire version while keeping the minimum supported version the same.